### PR TITLE
sink/mysql: adjust producer/consumer exit sequence to avoid goroutine leak

### PR DESCRIPTION
**Pull request by amyangfei**
### What problem does this PR solve? 

Fix a goroutine leak bug, where the sink worker has exited, but more txns are sent to txn channel of sink worker.
And the producer will wait for the txn to be consumed forever.

### What is changed and how it works?

Add a cleanup mechanism for sink worker.

1. producer(sink.flushRowChangedEvents goroutine) of txnCh exits
2. goroutine in 1 sends notification to closedCh of each sink worker
3. each sink worker receives the notification from closedCh and mark FinishWg as Done

### Check List 

Tests 

 - Unit test
 - Integration test

### Release note

- Fix a bug that some MySQL connection could leak after MySQL sink meets error and pauses.

### CODE REVIEW

**Review by liuzix — APPROVED** [review]

File: cdc/sink/mysql.go [hunk 1]
```diff
@@ -134,6 +134,11 @@ func (s *mysqlSink) FlushRowChangedEvents(ctx context.Context, resolvedTs uint64
 }
 
 func (s *mysqlSink) flushRowChangedEvents(ctx context.Context, receiver *notify.Receiver) {
+	defer func() {
+		for _, worker := range s.workers {
+			worker.closedCh <- struct{}{}
```
Comment (leoppro) [inline cdc/sink/mysql.go:139]: will the goroutine hang at L139 when the flushRowChangedEvents function return

**Review by leoppro — APPROVED** [review]

**Pull request merged**